model-test/src/main/java/org/jboss/as/model/test/ModelTestKernelServices.java

```diff
@@ -183,4 +189,43 @@ TransformedOperation transformOperation(ModelVersion modelVersion, ModelNode ope
 
     ImmutableManagementResourceRegistration getRootRegistration();
 
+    /**
+     * Get information about error on boot from both BootErrorCollector and BootError
+     *
+     * @return BootErrorCollector description if any and BootError printed stack trace, empty string otherwise
+     */
+    default String getBootErrorDescription(){
+        String errorDescription = "";
+        ModelNode result = getBootErrorCollectorFailures();
+        if (hasBootErrorCollectorFailures()) {
+            errorDescription += "BootErrorCollector failures: " + result.asString() + System.lineSeparator();
+        }
+        if (getBootError() != null) {
+            errorDescription += "BootError: " + getBootError().toString() + getBootError().getStackTrace().toString();
+        }
+        return errorDescription;
+    }
+
+    /**
+     *
+     * @return failures from BootErrorCollector otherwise null
+     */
+    default ModelNode getBootErrorCollectorFailures(){
+        ModelNode readBootErrorsOp = Util.createOperation("read-boot-errors", PathAddress.pathAddress(PathElement.pathElement(CORE_SERVICE, MANAGEMENT)));
+        ModelNode result = null;
+        try {
+            result = executeForResult(readBootErrorsOp);
+        } catch (OperationFailedException e) {
+            System.err.println("Error getting BootErrorCollector failures: " + e.toString());
+        }
+        return result;
+    }
+
+    /**
+     *
+     * @return true if BootErrorCollector collected any failure
+     */
+    default boolean hasBootErrorCollectorFailures() {
+        return getBootErrorCollectorFailures() != null && !getBootErrorCollectorFailures().asString().equals("undefined");
```

Comment:
Just in relation to the above, I'm wondering if we should remove `getBootErrorCollectorFailures().asString().equals("undefined")` here and make `getBootErrorCollectorFailures()` return null if only if there are no error failures. 

Just wondering, if you think we can leave it as it is, I'm fine, but it should be reflected on the Java Doc.

Reply:
Of course, you're right. :) There is no point in keeping undefined failure without any further details so I changed it as you suggested. Also, I improved the Java Doc. Hopefully, it looks more suitable now. What do you think? Is it okay like that or did I miss something?

Reply:
Thanks, @khermano , looks better now, merged